src/routes/commercialRegistrations/search.ts

```diff
@@ -0,0 +1,54 @@
+import type { Request, Response } from "express";
+import { businesses } from "../../fixtures/businesses";
+import { ModelNotFoundError } from "./types/modelNotFoundError";
+import { SearchQuery } from "./types/searchQuery";
+import { Registration } from "./types/registration";
+
+/**
+ * @see {@link https://docs.solarisgroup.com/api-reference/onboarding/businesses/#tag/Business-Registrations/paths/~1v1~1commercial_registrations~1search_by_name/get}
+ * Note: This endpoint may be used free of charge.
+ *
+ * Returns some business as if it exists
+ * Returns a mock business if more than 2 characters are provided for country and name
+ */
+export const search = async (
+  req: Request<{}, {}, {}, SearchQuery>,
+  res: Response<Registration[] | ModelNotFoundError>
+) => {
+  const { country = "DE", name } = req.query;
+  const foundBusinesses = businesses.filter(
+    (business) =>
+      String(business.name)
+        .toLowerCase()
+        .includes(String(name).toLowerCase()) &&
+      business.address.country === country
+  );
+  const highEffort = String(country).length + String(name).length > 2;
+
+  if (foundBusinesses) {
+    return res.status(200).send(
+      foundBusinesses.map((business) => ({
```

Comment:
could you please also add a list of random companies on mocksolaris and add search string to the generated company name, that it gets closer to the reality. do you know what I mean?
that we do not need to enter the same string every time on FE to see a list with many suggestions
We discussed it during grooming but unfortunately haven't added it to the ticket

Reply:
I've added some generation mechanism and tests

Reply:
it should also return the name

Reply:
True: https://github.com/kontist/mock-solaris/pull/197/commits/75201bba2c803fadf50223c9172ec69f1f86a764